apps/test/unit/signUpFlow/FinishTeacherAccountTest.tsx

```diff
@@ -1,154 +1,106 @@
-import {render, screen, fireEvent} from '@testing-library/react';
+import '@testing-library/jest-dom';
+import {fireEvent, render, screen} from '@testing-library/react';
 import React from 'react';
 
 import FinishTeacherAccount from '@cdo/apps/signUpFlow/FinishTeacherAccount';
 import locale from '@cdo/apps/signUpFlow/locale';
 import {
   DISPLAY_NAME_SESSION_KEY,
   EMAIL_OPT_IN_SESSION_KEY,
-  SCHOOL_ID_SESSION_KEY,
-  SCHOOL_ZIP_SESSION_KEY,
-  SCHOOL_NAME_SESSION_KEY,
 } from '@cdo/apps/signUpFlow/signUpFlowConstants';
-import i18n from '@cdo/locale';
+
+// mocking SchoolDataInputs as it has its own tests
+jest.mock('@cdo/apps/templates/SchoolDataInputs', () => () => (
+  <div>Mocked SchoolDataInputs</div>
+));
 
 describe('FinishTeacherAccount', () => {
-  afterEach(() => {
-    [
-      DISPLAY_NAME_SESSION_KEY,
-      EMAIL_OPT_IN_SESSION_KEY,
-      SCHOOL_ID_SESSION_KEY,
-      SCHOOL_ZIP_SESSION_KEY,
-      SCHOOL_NAME_SESSION_KEY,
-    ].forEach((session_key: string) => {
-      sessionStorage.removeItem(session_key);
-    });
+  beforeEach(() => {
+    sessionStorage.clear();
   });
 
-  function renderDefault(usIp: boolean = true) {
-    render(<FinishTeacherAccount usIp={usIp} />);
-  }
-
-  it('renders finish teacher account page with school zip when usIp is true', () => {
-    renderDefault(true);
-
-    // Renders page title
-    screen.getByText(locale.finish_creating_teacher_account());
-
-    // Renders questions shown regardless of usIp
-    screen.getByText(locale.what_do_you_want_to_be_called());
-    screen.getByText(i18n.whatCountry());
-    screen.getByText(locale.keep_me_updated());
-
-    // Renders school zip and select school questions if usIp is true
-    screen.getByText(i18n.enterYourSchoolZip());
-    screen.getByText(i18n.selectYourSchool());
-    expect(screen.queryByText(i18n.schoolOrganizationQuestion())).toBe(null);
-
-    // Renders button that finishes sign-up
-    screen.getByText(locale.go_to_my_account());
+  it('renders the finish teacher account page', () => {
+    render(<FinishTeacherAccount usIp={true} />);
+
+    expect(
+      screen.getByText(locale.finish_creating_teacher_account())
+    ).toBeInTheDocument();
```

Comment:
Nit: I'm not sure these need to be updated since `getByText` will throw an error if the given text is not found?

Reply:
ooo good question, I'll take a look!

Reply:
I _think_ this is okay. if `getByText` throws an error it should fail the test, which is what we would expect

Reply:
Yep it does! So I think if it's okay could we keep the changes to these tests what is specific to the changes made in this PR? Hannah and I have a couple PR's open that are touching several of these files so as we add more tests and such, if we are changing the structure/patterns used it will create a ton of merge conflicts that I'm worried will accidentally sweep something under the rug that we don't mean to. I think these are good future changes/cleanups that would be great to discuss but if possible I'd like to keep the scope of this PR to a minimum so it's easier to keep moving forward with other signup changes!

Reply:
yep, that's fair! I'll spend some time tomorrow morning trying to scope this down, though it will be tricky if we're trying to get away from relying on jquery and session storage to access the form state from all over the place. I'm happy to jump on a call to discuss the changes in more detail since I absolutely did not want to step on toes or disrupt ongoing work! in fact it would be really great to get a tour of other places this component is planned to be used. I definitely should have asked for that before I got too deep on this! 😅

Reply:
ah, I misunderstood your first point. I was confused about not seeing expect wrapping those `getByText`. I get it now, makes sense! I'll put this back the way I found it 😬

Reply:
Thank you! No for sure, thank you for your insight and I think this would be a great thing to discuss and cleanup soon! Definitely didn't step on any toes :D

Reply:
hoho, I had kind of forgotten why I even touched these tests in the first place! it was late on friday and I was getting frustrated with a json circular reference error I was getting instead of a test failure when running the first two tests together (or the whole suite). when run individually, each test passed fine.

the circular reference error was a red herring I'm still a bit confused about. but ultimately it was finding the zip code element with queryByText instead of null. the issue was `SCHOOL_COUNTRY_SESSION_KEY` was not being reset in sessionStorage in the afterEach block, which was taking precedence over the usIp prop. 

anyway, should be all good now!

Reply:
Ah gotcha gotcha!